inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp

```diff
@@ -0,0 +1,133 @@
+        auto jit = CreateJit(kernelName, cldnn_jit, entry_point);
+        auto& kernel = kd.kernels[0];
+        FillCLKernelData(kernel, runInfo.stage_1, params.engineInfo, kernelName, jit, entry_point);
+        kernel.arguments.clear();  // Clear original output argument
+        kernel.arguments.push_back({ArgumentDescriptor::Types::INPUT, 0});
+        kernel.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kd.internalBufferSizes.push_back(newParams.output.PhysicalSizeInBytes());
+    }
+    {
+        // Final
+        auto entry_point = GetEntryPoint(kernelName, newParams.layerID, options);
+        auto cldnn_jit = GetJitConstants(newParams, runInfo.stage_final);
+        std::string jit = CreateJit(kernelName, cldnn_jit, entry_point);
+
+        auto& kernel = kd.kernels[1];
+
+        FillCLKernelData(kernel, runInfo.stage_final, params.engineInfo, kernelName, jit, entry_point);
+
+        kernel.arguments.clear();  // Clear original output argument
+        kernel.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+    }
+    kd.intenralBufferDataType = Datatype::F32;
+    kd.estimatedTime = estimated_time;
+
+    return {kd};
+}
+
+CumSumKernelPartialSum::MultiDispatchData CumSumKernelPartialSum::SetDefaultForMulti(const cum_sum_params& params) const {
+    MultiDispatchData md;
+    std::vector<size_t> dims = {params.output.Batch().v,
+                                params.output.Feature().v,
+                                params.output.W().v,
+                                params.output.Z().v,
+                                params.output.Y().v,
+                                params.output.X().v};
+
+    size_t index = GetRealAxisIndex(params);
+    auto items_num = dims[index];
+
+    std::vector<size_t> gws(3, 0);
+    gws[0] = dims[index];
+    for (size_t i = 0, gws_i = 1; i < dims.size(); ++i) {
+        if (i == index)
+            continue;
+        if (gws[gws_i] == 0) {
+            gws[gws_i] = dims[i];
+        } else {
+            gws[gws_i] *= dims[i];
+            if (gws_i == 1)
+                ++gws_i;
+        }
+    }
+
+    md.stage_1.gws0 = Align(gws[0], BLOCK_SIZE);
+    md.stage_1.gws1 = gws[1];
+    md.stage_1.gws2 = gws[2];
+    md.stage_1.lws0 = BLOCK_SIZE;
+    md.stage_1.lws1 = 1;
+    md.stage_1.lws2 = 1;
+    md.stage_1.sum_items_num = items_num;
+
+    md.stage_final.gws0 = gws[0];
+    md.stage_final.gws1 = gws[1];
+    md.stage_final.gws2 = gws[2];
+    md.stage_final.lws0 = 1;
+    md.stage_final.lws1 = 1;
+    md.stage_final.lws2 = 1;
+    md.stage_final.sum_items_num = Align(items_num, BLOCK_SIZE);
+
+    return md;
+}
+
+KernelsData CumSumKernelPartialSum::GetKernelsData(const Params& params, const optional_params& options) const {
+    return GetMultiStageKernelsData(params, options, FORCE_PRIORITY_9);
```

Comment:
Why did you set so low priority for you kernel? Did you test your implementation on real topologies or only by unit-tests?

Reply:
Which priority would you prefer? I tested implementation by unit-tests and I know that it should be possible to improve performance for this operation. This is why I set only `FORCE_PRIORITY_9`.

Reply:
I suppose that 6-7 is a right choice if you have visible increase of performance in comparison with ref kernel. But this is only my mind.